### ISSUE TITLE
yabridge-3.8.1 fails to build

### ISSUE THREAD

**Issue by GovanifY** [NixOS/nixpkgs#179724]
Yabridge fails to build on my system due to an incorrect definition of open_sockets in winegcc, as shown in the build log below.

### Steps To Reproduce

Build yabridge in a recent system-env.

### Build log
```
[344/352] Compiling C++ object yabridge-group.p/src_wine-host_bridges_group.cpp.o
FAILED: yabridge-group.p/src_wine-host_bridges_group.cpp.o
wineg++ -Iyabridge-group.p -I. -I.. -I../src/include -Isrc/common/config -I../src/common/config -I../subprojects/bitsery/include -I../subprojects/function2/include -I../subprojects/tomlplusplus/include -I/nix/store/rcdw0mgn2wjcrnpjxzqr9avfvdigsph8-boost-1.79.0-dev/include -I/nix/store/w7s78mfv61lpq1sgizh8x3jwvvsrx5dy-libxcb-1.14-dev/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -Wextra -Wpedantic -std=c++2a -fvisibility=hidden -fvisibility-inlines-hidden -msse2 -DRELEASE=1 -m64 -DBOOST_FILESYSTEM_DYN_LINK=1 -DBOOST_ALL_NO_LIB -fvisibility=hidden -fvisibility-inlines-hidden -DBOOST_ASIO_DISABLE_CONCEPTS -DBOOST_POSIX_HAS_VFORK=1 -msse2 -DWITH_BITBRIDGE -DWITH_VST3 -DNOMINMAX -D__WINE_WINSOCKAPI_STDLIB_H -D_WINSOCKAPI_ -D__IFileOperation_INTERFACE_DEFINED__ -D__WINE_SAL_H__ -m64 -isystem../subprojects/vst3 -isystemsubprojects/vst3 -MD -MQ yabridge-group.p/src_wine-host_bridges_group.cpp.o -MF yabridge-group.p/src_wine-host_bridges_group.cpp.o.d -o yabridge-group.p/src_wine-host_bridges_group.cpp.o -c ../src/wine-host/bridges/group.cpp
../src/wine-host/bridges/group.cpp: In function 'boost::asio::local::stream_protocol::acceptor create_acceptor_if_inactive(boost::asio::io_context&, boost::asio::local::stream_protocol::endpoint&)':
../src/wine-host/bridges/group.cpp:299:35: error: variable 'std::ifstream open_sockets' has initializer but incomplete type
  299 |         std::ifstream open_sockets("/proc/net/unix");
      |                                   ^
winegcc: /nix/store/yg57yaaqchg62sxs4fhj6wxn8q3mjd8b-gcc-wrapper-11.3.0/bin/g++ failed
ninja: build stopped: subcommand failed.
error: builder for '/nix/store/awy3111v9f9psimcc4k4a98sai6y7gbx-yabridge-3.8.1.drv' failed with exit code 1
```

### Notify maintainers
@kira-bruneau 

### Metadata

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.35-hardened1, NixOS, 22.05 (Quokka)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.7.0`
 - channels(govanify): `"home-manager, nixos-mailserver, nixpkgs"`
 - channels(root): `"home-manager, nixos, nixos-mailserver"`
 - nixpkgs: `/home/govanify/.nix-defexpr/channels/nixpkgs`

```

**Comment by jpotier** [comment]
Tried build with sandbox disabled, but same issue.

**Comment by SFrijters** [comment]
Probably worth bumping the package to version 4.x? The changelog mentions dropping boost entirely.

**Comment by nyarla** [comment]
This problem is incompatible version of boost.

yabridge 3.8.1 can compile with boost175 in my environment, but it can't build with latest version.

I use this workaround for this issue:

```nix
    yabridge = pkgs.yabridge.override {
      # temporary fix for 3.8.1
      boost = pkgs.boost175;
      pkgsi686Linux = {
        boost = pkgs.pkgsi686Linux.boost175;
        xorg.libxcb = pkgs.pkgsi686Linux.xorg.libxcb;
      };
    };
```
